src/Generating/StructGen.cpp

```diff
@@ -183,6 +183,14 @@ void cStructGenTrees::ApplyTreeImage(
 				CASE_TREE_OVERWRITTEN_BLOCKS:
 				{
 					a_ChunkDesc.SetBlockTypeMeta(itr->m_RelX, itr->m_RelY, itr->m_RelZ, itr->m_BlockType, itr->m_BlockMeta);
+					// If grass is below our tree, turn it to dirt
+					if (
+						(cBlockInfo::IsSolid(itr->m_BlockType)) &&
```

Comment:
Water can decay grass. But since this is a tree being generated on grass, there is no water here.